### PR TITLE
Trio

### PR DESCRIPTION
**Pull request by mehaase**
This PR adds support for the Trio async framework. https://trio.readthedocs.io/en/v0.9.0/index.html

(This includes the changes to the README from #72 because I wanted to add an example for this Trio mode, too, but I am not opposed to eliminating that.)

### CODE REVIEW

**Comment by gabor-boros** [comment]
@mehaase, Could you please fix the codacy issues?

**Comment by mehaase** [comment]
> @mehaase, Could you please fix the codacy issues?

Yes, I can certainly try, but I'm not certain how to read the Codacy results. In the overview above it says, "added 15". What does that mean? I tried following the link but it says no new issues.

screenshot from 2019-01-15 11-36-24

When I clicked the files tab, it says it can't find any files! Not sure what I'm doing wrong. ¯\_(ツ)_/¯

screenshot from 2019-01-15 11-20-42

**Comment by gabor-boros** [comment]
@mehaase and @grandquista I restarted the sync on codacy. Hopefully it will help to see everything. By the way, we should investigate why is this happening some times.

File: rethinkdb/trio_net/net_trio.py [hunk 1]
```diff
@@ -0,0 +1,507 @@
...
+
+    @classmethod
+    def open(cls, *args, **kwargs):
+        return cls(*args, **kwargs)
+
+    async def __aenter__(self):
+        self._conn = await connect(*self._args, **self._kwargs)
+        return self._conn
+
+    async def __aexit__(self, self, exc_type, exc, traceback):
```
Comment (grandquista) [inline rethinkdb/trio_net/net_trio.py]: Double self here. Otherwise looks good to merge.

Comment (mehaase) [reply rethinkdb/trio_net/net_trio.py]: Ahh yes, I missed a few other issues, too. Just pushed a commit to fix those.

*Please keep this PR open for a bit longer!* I am still integrating it into my own app, and I'm not in a rush to have this merged.

Comment (gabor-boros) [reply rethinkdb/trio_net/net_trio.py]: @mehaase What do you mean by longer? Can we merge it when all the checks are passes?

**Comment by gabor-boros** [comment]
@mehaase Can I merge the PR? Or would you like to keep it opened?

**Comment by mehaase** [comment]
Please keep it open. I will write a few integration tests and I am also integrating it into an application I'm writing which will help me spot any undiscovered problems. I'll be ready to merge in about a week.

**Comment by gabor-boros** [comment]
@mehaase Ok, Then let us know when we can check the PR.

**Comment by gabor-boros** [comment]
@mehaase, do you have any update on this?

**Comment by mehaase** [comment]
Yes, I'm writing an app that is using the new driver in this PR, and I want to get that app working before we merge this, just to make sure I haven't missed anything obvious. I have noticed one little change to the docs that needs to be made. Other than that, it's working well so far.

I should be finished with this app I'm writing by middle of next week, so I will be ready to merge by Feb 22 at the latest.

**Comment by mehaase** [comment]
@gabor-boros @grandquista I'm ready to merge this now. I've been building a production application using this driver and I am satisfied with the features and stability. I just pushed a commit with one small change to the README and I am ready to merge this branch.

**Pull request merged**